### PR TITLE
Fix macOS home-brew PR slack notifications for new release

### PR DESCRIPTION
**Pull request by NlightNFotis**
Currently, the slack notification action we use to notify
us in case jobs fail doesn't work on macOS and Windows (because
of a docker issue bootstrapping an Alpine linux environment on
top of those, see https://github.com/rtCamp/action-slack-notify/issues/22).

This is allowing us to do that, by cloning the action source code
into our `scripts/` folder, and then running it using a downloaded
`golang` toolchain - instead of needing to run the whole action on
top of docker.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: scripts/slack_notification_action.go [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+	color := ""
+	switch os.Getenv(EnvSlackColor) {
+	case "success":
+		color = "good"
+	case "cancelled":
+		color = "#808080"
+	case "failure":
+		color = "danger"
+	default:
+		color = envOr(EnvSlackColor, "good")
```
Comment (TGWDB) [inline scripts/slack_notification_action.go]: Should the default really be "good"? Isn't this what we would see if we have some unexpected outcome (infrastructure unavailable?)... I would assume we'd like to see some warning or error, not that it worked.

File: scripts/slack_notification_action.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+		Channel:   os.Getenv(EnvSlackChannel),
+		LinkNames: os.Getenv(EnvSlackLinkNames),
+		Attachments: []Attachment{
+			{
+				Fallback:   envOr(EnvSlackMessage, "GITHUB_ACTION="+os.Getenv("GITHUB_ACTION")+" \n GITHUB_ACTOR="+os.Getenv("GITHUB_ACTOR")+" \n GITHUB_EVENT_NAME="+os.Getenv("GITHUB_EVENT_NAME")+" \n GITHUB_REF="+os.Getenv("GITHUB_REF")+" \n GITHUB_REPOSITORY="+os.Getenv("GITHUB_REPOSITORY")+" \n GITHUB_WORKFLOW="+os.Getenv("GITHUB_WORKFLOW")),
+				Color:      color,
+				AuthorName: envOr(EnvGithubActor, ""),
+				AuthorLink: os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv(EnvGithubActor),
+				AuthorIcon: os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv(EnvGithubActor) + ".png?size=32",
+				Footer:     envOr(EnvSlackFooter, "<https://github.com/rtCamp/github-actions-library|Powered By rtCamp's GitHub Actions Library>"),
```
Comment (TGWDB) [inline scripts/slack_notification_action.go]: Can we do this without a footer? If not, can we make the footer empty? I don't see a benefit in advertising/adding footers here. (The messages are already quite spammy...)

Comment (NlightNFotis) [reply scripts/slack_notification_action.go]: Yeah, happy to leave this empty.

File: scripts/slack_notification_action.go (hunk 1 above)
Comment (NlightNFotis) [reply scripts/slack_notification_action.go]: I'll see if I can find one that's a neutral colour (say, yellow or grey)

Comment (TGWDB) [reply scripts/slack_notification_action.go]: Or failure. We should assume the "bad" path here, not the "good" path.

**Review by TGWDB — APPROVED** [review]

**Pull request merged**